src/components/Posts/EditPostModal.tsx

```diff
@@ -0,0 +1,160 @@
+import {
+  Dialog,
+  DialogTitle,
+  Button,
+  DialogContent,
+  IconButton,
+  Box,
+  Stack,
+  TextField,
+  Typography,
+} from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+import { useEffect, useState } from "react";
+import { useAppDispatch, useAppSelector } from "../../state/hooks";
+import { EmojiClickData } from "emoji-picker-react";
+import EmojiPickerIconButton from "../Common/EmojiPickerIconButton";
+import UserAvatar from "../Common/UserAvatar";
+import AddPhotoAlternateOutlinedIcon from "@mui/icons-material/AddPhotoAlternateOutlined";
+import axios from "axios";
+import { updatePost } from "../../state/slices/postsSlice";
+
+type EditPostModalProps = {
+  onClose: () => void;
+  open: boolean;
+  postId: number;
+};
+
+const styles = {
+  paperProps: {
+    overflow: "visible",
+    borderRadius: 20,
+  },
+  dialog: {
+    ".MuiDialog-scrollPaper": { alignItems: "flex-start" },
+  },
+  dialogTitle: {
+    display: "flex",
+    alignItems: "center",
+  },
+  titleText: {
+    fontWeight: "bold",
+    paddingLeft: 3,
+  },
+  dialogContentContainer: {
+    display: "flex",
+    paddingBottom: 2,
+    paddingLeft: 2,
+    paddingRight: 2,
+    paddingTop: 1,
+  },
+  avatar: { paddingRight: 1.5 },
+  textFieldContainer: { width: "100%", paddingTop: 0.5 },
+  textField: { paddingBottom: 2 },
+  postActions: {
+    display: "flex",
+    justifyContent: "space-between",
+    marginTop: 1,
+  },
+};
+
+const EditPostModal = ({ onClose, open, postId }: EditPostModalProps) => {
+  const user = useAppSelector((state) => state.user);
+  const dispatch = useAppDispatch();
+  const existingPost = useAppSelector((state) =>
+    state.posts.posts.find((post) => post.postId === postId)
+  );
+  const [postTextContent, setPostTextContent] = useState<string>("");
+
+  useEffect(() => {
+    if (existingPost) {
+      setPostTextContent(existingPost.textContent);
+    }
+  }, [existingPost]);
+
+  const onSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!existingPost) {
+      throw new Error(`Post with ID ${postId} not found`);
+    }
+
+    try {
+      await axios.put("http://localhost:3001/api/posts/editPost", {
+        postId: postId,
+        textContent: postTextContent,
+      });
+      dispatch(
+        updatePost({
+          ...existingPost!,
+          textContent: postTextContent,
+        })
+      );
```

Comment:
You also need to update the `editedTimestamp` here

Reply:
Done

Reply:
No, like I meant on the front end. I've done it for you though

Reply:
Ahh, I didnt think i had to use the `Tostring()` for the date